components/AddTask.js

```diff
@@ -1,17 +1,72 @@
+import { useRef } from "react";
+import axios from '../utils/axios';
+import {useAuth} from '../context/auth';
+import Toast from './Toast';
+import {useToast} from '../context/toast';
+
 export default function AddTask() {
+  const taskRef = useRef(null);
+  const {token} = useAuth();
+  const [state, show, text, spinner, setState, setShow, setText, setSpinner, hideToast] = useToast();
+
   const addTask = () => {
-    /**
-     * @todo Complete this function.
-     * @todo 1. Send the request to add the task to the backend server.
-     * @todo 2. Add the task in the dom.
-     */
+     const title = taskRef.current.value;
+
+     if (title != "") {
+      setShow(true);
+      setState('neutral');
+      setText("Adding Task");
+      setSpinner(true);
+
+       const data = {
+         title,
+       };
+
+       axios({
+         url: "todo/create/",
+         method: "post",
+         headers: {
+           Authorization: `Token ${token}`,
+         },
+         data,
+       })
+         .then(({ data, status }) => {
+           const event = new CustomEvent("addedTask", {
+             bubbles: true,
+           });
+           taskRef.current.dispatchEvent(event);
+
+           setState('success');
+           setText("Task Added Successfully");
+           setSpinner(false);
+           hideToast();
+
```

Comment:
You should clear the input Field after successfully adding a new ToDo Item

Reply:
I have fixed this now. May be reviewed again...